FinancialTracker/Firebase.swift

```diff
@@ -66,20 +72,49 @@ class FirebaseHandler {
     }
 
     func signIn(email: String, password: String, completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
-        Auth.auth().signIn(withEmail: email, password: password) { _, error in
+        auth.signIn(withEmail: email, password: password) { _, error in
 
-            if error != nil {
+            guard error == nil else {
                 completionHandler(FirebaseError.auth(error), false)
+                return
             }
             completionHandler(nil, true)
         }
     }
     
-    func signOut() {
+    func signOut(completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
         do {
-            try Auth.auth().signOut()
+            try auth.signOut()
         } catch let signOutError {
-            print("Error signing out: %@", FirebaseError.signOut(signOutError))
+            completionHandler(FirebaseError.signOut(signOutError), false)
+        }
+    }
+    
+    func getCurrentUserData(completionHandler: @escaping (FirebaseError?, Bool, [String: Any]?) -> Void) {
+        guard let uid = auth.currentUser?.uid else {
+            fatalError("Can't access without logged in user.")
+        }
+        
+        database.collection(DBCollectionKey.users.rawValue).document(uid).getDocument(completion: { document, error in
+            guard error == nil else {
+                completionHandler(FirebaseError.database(error), false, nil)
+                return
+            }
+            
+            guard let document = document else {
+                completionHandler(FirebaseError.unknown, false, nil)
+                return
+            }
+                        
+            completionHandler(nil, true, document.data())
+        })
+    }
+    
+    func addDataToDocument(collection: String, data: [String: Any]) {
```

Comment:
We do not want to see strings flying around in functions. 
Try to give strings types where possible - use enums, structs, typealias for that - up to you.